### PR TITLE
update rerank documentation

### PR DESCRIPTION
**Pull request by dtrawins**
### 🧪 Checklist

- [ ] Unit tests added.
- [x] The documentation updated.
- [ ] Change follows security best practices.

https://openvino-doc.iotg.sclab.intel.com/rerank/openvino-workflow/model-server/ovms_docs_clients_genai.html

### CODE REVIEW

File: demos/rerank/README.md [hunk 1]
```diff
@@ -16,142 +16,85 @@ It will create an image called `openvino/model_server:latest`.
...
-pip3 install optimum-intel@git+https://github.com/huggingface/optimum-intel.git  openvino-tokenizers[transformers]==2024.4.* openvino==2024.4.* nncf>=2.11.0 sentence_transformers==3.1.1 openai "transformers<4.45"
+pip3 install -r https://raw.githubusercontent.com/openvinotoolkit/model_server/refs/heads/export-models-script/demos/common/export_models/requirements.txt
 ```
 
 Run optimum-cli to download and quantize the model:
 ```bash
-cd demos/rerank
-convert_tokenizer -o models/BAAI/bge-reranker-large-tokenizer/1 BAAI/bge-reranker-large
-optimum-cli export openvino --disable-convert-tokenizer --model BAAI/bge-reranker-large --task text-classification --trust-remote-code models/BAAI/bge-reranker-large-rerank/1
+wget https://raw.githubusercontent.com/openvinotoolkit/model_server/refs/heads/export-models-script/demos/common/export_models/export_model.py
```
Comment (dkalinowski) [inline demos/rerank/README.md:27]: Branch will be deleted

Comment (dkalinowski) [reply demos/rerank/README.md:27]: @kblaszczak-intel what should we specify here to get sphinx documentation to work?

Main branch? - if so, when we create release branch we will refer to incompatible script, way newer than the one release documentation refers to

File: demos/rerank/README.md [hunk 2]
```diff
@@ -16,142 +16,85 @@ It will create an image called `openvino/model_server:latest`.
 > **Note** Python 3.9 or higher is needed for that step
 Here, the original Pytorch LLM model and the tokenizer will be converted to IR format and optionally quantized.
 That ensures faster initialization time, better performance and lower memory consumption.
-LLM engine parameters will be defined inside the `graph.pbtxt` file.
 
 Install python dependencies for the conversion script:
 ```bash
-export PIP_EXTRA_INDEX_URL="https://download.pytorch.org/whl/cpu"
-pip3 install optimum-intel@git+https://github.com/huggingface/optimum-intel.git  openvino-tokenizers[transformers]==2024.4.* openvino==2024.4.* nncf>=2.11.0 sentence_transformers==3.1.1 openai "transformers<4.45"
+pip3 install -r https://raw.githubusercontent.com/openvinotoolkit/model_server/refs/heads/export-models-script/demos/common/export_models/requirements.txt
```
Comment (dkalinowski) [inline demos/rerank/README.md:22]: Branch will be deleted

File: demos/rerank/README.md [hunk 3]
```diff
@@ -16,142 +16,85 @@ It will create an image called `openvino/model_server:latest`.
...
-```python
-from langchain_community.document_compressors.openvino_rerank import OpenVINOReranker, RerankRequest
+```bash
+python compare_results.py --query "hello" --document "welcome" --document "farewell"
+query hello
+documents ['welcome', 'farewell']
+HF Duration: 145.731 ms
+OVMS Duration: 23.227 ms
+HF reranking: [0.99640983 0.08154089]
+OVMS reranking: [0.9968273 0.0913821]
```
Comment (dkalinowski) [inline demos/rerank/README.md:92]: Why does it differ? It was exactly the same during my tests. Isn't it different model compression?

File: docs/model_server_rest_api_chat.md [hunk 4]
```diff
@@ -65,10 +65,10 @@ curl http://localhost/v3/chat/completions \
...
-| stream | ✅ | ✅ | ✅ | bool (optional, default: `false`) | If set to true, partial message deltas will be sent to the client. The generation chunks will be sent as data-only [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format) as they become available, with the stream terminated by a `data: [DONE]` message. [Example Python code](clients_openai.md) |
+| stream | ✅ | ✅ | ✅ | bool (optional, default: `false`) | If set to true, partial message deltas will be sent to the client. The generation chunks will be sent as data-only [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format) as they become available, with the stream terminated by a `data: [DONE]` message. [Example Python code](clientsgenai.md) |
```
Comment (dkalinowski) [inline docs/model_server_rest_api_chat.md:68]:
```suggestion
| stream | ✅ | ✅ | ✅ | bool (optional, default: `false`) | If set to true, partial message deltas will be sent to the client. The generation chunks will be sent as data-only [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format) as they become available, with the stream terminated by a `data: [DONE]` message. [Example Python code](clients_genai.md) |
```

Comment (dkalinowski) [reply docs/model_server_rest_api_chat.md:68]: The same error in all other pages

File: demos/rerank/README.md (hunk 3 above)
Comment (dtrawins) [reply demos/rerank/README.md:92]: yes, this is int8 quantized model

File: demos/rerank/README.md (hunk 2 above)
Comment (dtrawins) [reply demos/rerank/README.md:22]: after branch export-models-script is merged, it will be updated to main

File: demos/rerank/README.md (hunk 1 above)
Comment (dtrawins) [reply demos/rerank/README.md:27]: @dkalinowski on nightly build we would use main branch, on 2024 version it will be releases/2024/5

Comment (dkalinowski) [reply demos/rerank/README.md:27]: What if we split this script to multiple files in the future? Downloading this single script will cause errors. The script exists within the export_models directory with all the neccessary files like requirements.txt.
I think we should add instruction how to enter this directory and run installation scripts within this directory

File: demos/rerank/README.md (hunk 3 above)
Comment (dkalinowski) [reply demos/rerank/README.md:92]: Shouldn't we showcase that accurracy is 100%?

File: demos/rerank/README.md (hunk 1 above)
Comment (kblaszczak-intel) [reply demos/rerank/README.md:27]: I can confirm what Dariusz said. If you need help in turning all links to 24.5 when the branch is ready, let me know. We can help. I am not sure what other implications derive from it but I'm pretty sure we won't be able to change how links are handled in docs any time soon. Unfortunately.

**Review by michalkulakowski — APPROVED** [review]

File: demos/rerank/README.md [hunk 5]
```diff
@@ -16,142 +16,92 @@ It will create an image called `openvino/model_server:latest`.
...
-# Handle the response
-if response.status_code == 200:
-    reranked_results = response.json()
-    print("Reranked Results:", reranked_results)
-else:
-    print("Error:", response.status_code, response.text)
+## Comparison with Hugging Faces
 
+```bash
+python compare_results.py --query "hello" --document "welcome" --document "farewell"
```
Comment (porlows1) [inline demos/rerank/README.md:86]: To run it in automatic tests we need to reference this file as i.e. demos/rerano/compare_results.py

**Review by dkalinowski — APPROVED** [review]

File: docs/clients_genai.md [hunk 6]
```diff
@@ -8,43 +8,32 @@ hidden:
...
-Beside Tensorflow Serving API and KServe API frontends, the model server has now option to delegate the REST input deserialization and output serialization to a MediaPipe graph. A custom calculator can implement any form of REST API including streaming based on [Server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events).
+Beside Tensorflow Serving API (`/v1`) and KServe API (`/v2`) frontends, the model server supports a range of endpoints for generative use cases (`v3`). They are extendible using MediaPipe graphs.
+Currently supported endpoints are:
 
-We are introducing OpenAI compatible endpoints:
+OpenAI compatible endpoints:
 - [chat/completions](./model_server_rest_api_chat.md)
 - [completions](./model_server_rest_api_completions.md).
 - [embeddings](./model_server_rest_api_embeddings.md)
+Cohere Compatible endpoint:
```
Comment (porlows1) [inline docs/clients_genai.md:22]: Please fix formatting (additional newline?)

File: docs/clients_genai.md [hunk 7]
```diff
@@ -8,43 +8,32 @@ hidden:
...
 ```
 ## Introduction
-Beside Tensorflow Serving API and KServe API frontends, the model server has now option to delegate the REST input deserialization and output serialization to a MediaPipe graph. A custom calculator can implement any form of REST API including streaming based on [Server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events).
+Beside Tensorflow Serving API (`/v1`) and KServe API (`/v2`) frontends, the model server supports a range of endpoints for generative use cases (`v3`). They are extendible using MediaPipe graphs.
+Currently supported endpoints are:
 
-We are introducing OpenAI compatible endpoints:
+OpenAI compatible endpoints:
 - [chat/completions](./model_server_rest_api_chat.md)
 - [completions](./model_server_rest_api_completions.md).
```
Comment (porlows1) [inline docs/clients_genai.md:20]:
```suggestion
- [completions](./model_server_rest_api_completions.md)
```

**Review by porlows1 — APPROVED** [review]

**Pull request merged**